V0.12.0 rc

rhornung67 opened the pull request:
# Summary

- This PR merges develop plus changes in v0.12.0-rc branch into main so we can tag main for the release.
- Note that there are many changes. The differences between develop and v0.12.0-rc are much smaller and this is what should be reviewed as all other content in develop has been approved in numerous PRs. However, please approve this PR based on those changes. Here is the link to the PR with the changes between develop and v0.12.0-rc: https://github.com/LLNL/RAJA/pull/876
- After this is merged into main, the release tag will be created and a PR will be put up to merge back to develop.

gzagaris commented:
At last, the wait is (almost) over! 🎉

rchen20 commented on RELEASE_NOTES.md:
```diff
@@ -10,6 +10,108 @@
 Version vxx.yy.zz -- Release date 20yy-mm-dd
 ============================================
```
Take this out?

```suggestion
```

rhornung67 replied:
That's always been in the file. It's a placeholder for starting the next release.

mdavis36 commented on RELEASE_NOTES.md:
```diff
@@ -10,6 +10,108 @@
+        can wrap multiple arrays so their accesses can share index arithmetic.
+      * Introduced RAJA "Teams" concept as an experimental feature. This
+        enables hierarchical parallelism and additional nested loop patterns
+        beyond what RAjA::kernel supports. Please note that this is very much
```
RAjA -> RAJA

rhornung67 replied:
Thanks @mdavis36. Will fix.

rhornung67 commented:
@mdavis36 @rchen20 @davidbeckingsale @MrBurmark I addressed all the typos you reported and updated some build scripts that now work with CUDA 11. Please review if you want to take another look before this is merged. Thank you.

rchen20 commented on RELEASE_NOTES.md:
```diff
@@ -10,6 +10,108 @@
+  * Notable repository change:
+      * The 'master' branch in the RAJA git repo has been renamed to 'main'.
+
+  * New features:
```
@rhornung67 @MrBurmark Did we want to make users aware of "sort" as a New Feature? Maybe a change like the following (just the first sentence from the sort documentation):
```suggestion
      * New RAJA "sort" capability added. This provides portable parallel sort operations, which are basic parallel algorithm building blocks.
```

rhornung67 replied:
I thought I had it in there. Sorry. It is now added.
